python/sanlock.c

```diff
@@ -1392,16 +1392,20 @@ unregister the event listener using end_event.");
 static PyObject *
 py_reg_event(PyObject *self __unused, PyObject *args)
 {
-    const char *lockspace = NULL;
+    PyObject *lockspace = NULL;
     int fd = -1;
 
-    if (!PyArg_ParseTuple(args, "s", &lockspace))
+    if (!PyArg_ParseTuple(args, "O&", convert_to_pybytes, &lockspace)) {
+        Py_XDECREF(lockspace);
         return NULL;
+    }
 
     Py_BEGIN_ALLOW_THREADS
-    fd = sanlock_reg_event(lockspace, NULL /* event */, 0 /* flags */);
+    fd = sanlock_reg_event(PyBytes_AsString(lockspace), NULL /* event */, 0 /* flags */);
     Py_END_ALLOW_THREADS
 
+    Py_DECREF(lockspace);
```

Comment:
Why not add proper cleanup here like in previous patches?

We can do:

```
    if (fd < 0) {
        __set_exception(fd, "Unable to register event fd");
       goto finally;
    }

finally:
    Py_XDECREF(lockspace);

    if (fd < 0)
        return NULL;

    return Py_BuildValue("i", fd);
```

Reply:
We can. But it will larger then actual cleanup to be done in this case. It will be only for lockspace. We can always add finaly label when time will come to add more cleanups.

Reply:
Not sure what is "larger then actual cleanup". Why not keep the code consistent?